vault/provider_test.go

```diff
@@ -618,54 +618,131 @@ func TestAccTokenName(t *testing.T) {
 							}
 						}
 					},
-					Config: testTokenNameConfig(test.UseTokenNameSchema, test.TokenNameSchema),
-					Check:  testTokenName_check(test.WantTokenName),
+					Config: testProviderConfig(test.UseTokenNameSchema, `token_name = "`+test.TokenNameSchema+`"`),
+					Check:  checkTestSecretData("display_name", test.WantTokenName),
 				},
 			},
+			CheckDestroy: testEnvCleanup("VAULT_TOKEN_NAME"),
+		})
+	}
+}
+
+func TestAccChildToken(t *testing.T) {
+
+	tests := []struct {
```

Comment:
I think that we probably only need the `ExpectChildToken` and `TestChildTokenEnv` fields, all of the others can be dropped.

You can use `strconv.FormatBool(test.ExpectChildToken)` for use in the string context.

See:  https://pkg.go.dev/strconv#FormatBool for more details

Reply:
I don't think so.
There are two configuration methods being tested in the parameterization: the schema (provider config block in .tf file) and/or the environment variables that manage the same configuration. Because we want to be able to test them independently as well as together, we need a way to state whether we want to **use** the schema, the env var, both, or neither.

I have updated the names of these fields to be more clear that they set whether to **use** a particular config method, and then assign the `skip_child_token` setting separately. Please note that each of these fields is independent and not always the same bool as any other field.

Reply:
This simply followed a pattern already set by other tests in here like the existing `TestAccTokenName`. I'm certainly not going to remove valid test cases at this point - more coverage is better.

Reply:
Agreed, it's fine to keep it as-is for now.